### PR TITLE
qemu_guest_agent: new case to get os info of vm

### PR DESCRIPTION
**Pull request by xiagao**
Get operationg system info of vm via qga.
ID: 1813118
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
(01/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (74.47 s)
 (02/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (72.40 s)
 (03/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (117.74 s)
 (04/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (151.14 s)
 (05/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (80.13 s)
 (06/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (89.32 s)
 (07/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (95.72 s)
 (08/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (69.92 s)
 (09/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (60.57 s)
 (10/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (72.27 s)
 (11/11) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (91.13 s)

**Comment by xiagao** [comment]
update.
 (01/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (121.93 s)
 (02/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (124.90 s)
 (03/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (109.30 s)
 (04/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (131.67 s)
 (05/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (67.62 s)
 (06/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (386.79 s)
 (07/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (388.81 s)
 (08/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (95.80 s)
 (09/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (69.87 s)
 (10/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (62.02 s)
 (11/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (72.69 s)
 (12/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (70.00 s)

File: qemu/tests/qemu_guest_agent.py [hunk 1]
```diff
@@ -2686,6 +2686,125 @@ def bl_check(qga_cmd):
...
+            # for rhel8+ there is no variant info
+            # server or client
+            variant_qga = os_info_qga["variant"]
+            variant_id_qga = os_info_qga["variant-id"]
+            if "server" in os_name_full_guest.lower():
+                _result_check(variant_qga, "server")
+                _result_check(variant_id_qga, "server")
+            else:
+                _result_check(variant_qga, "client")
+                _result_check(variant_id_qga, "client")
```
[the remark this answers is not in the document]
Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py]: Ok, updated.

**Comment by xiagao** [comment]
@xiangchunfu  Call for your review...

File: qemu/tests/qemu_guest_agent.py [hunk 2]
```diff
@@ -2686,6 +2686,125 @@ def bl_check(qga_cmd):
...
+            kernel_g = re.search(r'(\d+\.\d+)\.(\d+)',
+                                 kernel_info_guest, re.I)
+            kernel_version_guest = kernel_g.group(1)
+            kernel_release_guest = kernel_g.group(2)
+        else:
+            kernel_version_guest = kernel_info_guest
+            cmd_get_kernel_rel = params["cmd_get_kernel_rel"]
+            kernel_release_guest = session.cmd_output(
+                cmd_get_kernel_rel).strip()
+        _result_check(kernel_version_qga, kernel_version_guest)
```
Comment (xiangchunfu) [inline qemu/tests/qemu_guest_agent.py:2781]: @xiagao currently,  'uname -v' print less than information for some OS.  Are you sure it is what you need?
# uname -v
#1 SMP Tue Feb 18 16:39:12 EST 2020

Comment (xiagao) [reply qemu/tests/qemu_guest_agent.py:2781]: yes, it is.
{"return": {"name": "Red Hat Enterprise Linux", "kernel-release": "4.18.0-176.el8.x86_64", "version": "8.2 (Ootpa)", "pretty-name": "Red Hat Enterprise Linux 8.2 Beta (Ootpa)", "version-id": "8.2", "kernel-version": "#1 SMP Wed Feb 5 00:30:18 UTC 2020", "machine": "x86_64", "id": "rhel"}}

Comment (xiangchunfu) [reply qemu/tests/qemu_guest_agent.py:2781]: Acked by: Xiangchun Fu xfu@redhat.com

**Comment by lolyu** [comment]
@xiagao please resolve conflict.

**Comment by xiagao** [comment]
@lolyu Update.
(01/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.8.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (78.02 s)
 (02/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (424.42 s)
 (03/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (425.65 s)
 (04/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (429.31 s)
 (05/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (388.40 s)
 (06/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2012.x86_64.r2.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (61.47 s)
 (07/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (88.80 s)
 (08/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: ERROR: VM creation command failed: PASS (108.81 s)
 (09/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (393.39 s)
 (10/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (71.52 s)
 (11/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (71.43 s)
 (12/12) Host_RHEL.m8.u2.product_av.qcow2.virtio_blk.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_os_info.q35: PASS (69.12 s)

**Comment by lolyu** [comment]
LGTM

**Pull request merged**